modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -305,14 +328,33 @@ public void run() {
                         // leaseholders at all.
                         if (isLeaseOutdated(lease)) {
                             // New lease is granting.
-                            writeNewLeaseInMetaStorage(grpId, lease, candidate);
+                            writeNewLeaseInMetaStorage(grpId, lease, candidate, renewedLeases, toBeNegotiated);
                         } else if (lease.isProlongable() && candidate.name().equals(lease.getLeaseholder())) {
                             // Old lease is renewing.
-                            prolongLeaseInMetaStorage(grpId, lease);
+                            prolongLeaseInMetaStorage(grpId, lease, renewedLeases);
                         }
                     }
                 }
 
+                byte[] renewedValue = new LeaseBatch(renewedLeases.values()).bytes();
+
+                var key = PLACEMENTDRIVER_LEASES_KEY;
+
+                msManager.invoke(
+                        or(notExists(key), value(key).eq(new LeaseBatch(leaseTracker.leasesCurrent()).bytes())),
```

Comment:
May be better to provide API to avoid extra serialization?

Reply:
what exactly do you suggest?

Reply:
I have rewritten the leases cache.